File: aoc_2022/ref_1.c
#include <stdint.h>
#include <stdbool.h>

typedef intptr_t isize;

isize doit(char *data, isize len) {
  isize sum, item, ipos, max;
  max = 0;
  ipos = 0;
  while (ipos < len) { // fyy
    sum = 0;
    while (ipos < len) {
      item = 0;
      const isize prebyte = data[ipos];
      const isize pretoken = prebyte-48;
      if (pretoken < 0 || pretoken > 9) break;
      while (ipos < len) { // fyyy
        const isize byte = data[ipos];
        const isize token = byte-48;
        if (token < 0 || token > 9) break;
        item = 10*item+token;
        ipos++;
      }
      sum += item;
      ipos++;
    }
    ipos++;
    if (sum > max) max = sum;
  }
  return max;
}

File: aoc_2022/ref1.zig
export fn doit(data: [*]u8, len: usize) callconv(.c) isize {
    var max: isize = 0;
    var ipos: usize = 0;
    while (ipos < len) { // fyy
        var sum: isize = 0;
        while (ipos < len) {
            var item: isize = 0;
            const prebyte: isize = data[ipos];
            const pretoken: isize = prebyte - 48;
            if (pretoken < 0 or pretoken > 9) break;
            while (ipos < len) { // fyyy
                const byte: isize = data[ipos];
                const token: isize = byte - 48;
                if (token < 0 or token > 9) break;
                item = 10 * item + token;
                ipos += 1;
            }
            sum += item;
            ipos += 1;
        }
        ipos += 1;
        if (sum > max) max = sum;
    }
    return max;
}
